examples_tests/61.BlitFilterTest/default_compute_normalization.comp

```diff
@@ -0,0 +1,192 @@
+#version 460 core
+
+#include <nbl/builtin/glsl/macros.glsl>
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_X_
+#define _NBL_GLSL_WORKGROUP_SIZE_X_ 8
+#endif
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_Y_
+#define _NBL_GLSL_WORKGROUP_SIZE_Y_ 8
+#endif
+
+#ifndef _NBL_GLSL_WORKGROUP_SIZE_Z_
+#define _NBL_GLSL_WORKGROUP_SIZE_Z_ 8
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_
+#define _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_ 256
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_
+#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ must be defined
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_FORMAT_ must be defined
+#endif
+
+// #define _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage1D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image1D
+	#endif
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage2D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image2D
+	#endif
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage3D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image3D
+	#endif
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_PIXEL_TYPE_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_PIXEL_TYPE_DEFINED_
+struct nbl_glsl_blit_normalization_pixel_t
+{
+	vec4 data;
+};
+#endif
+
+#define _NBL_GLSL_WORKGROUP_SIZE_ (_NBL_GLSL_WORKGROUP_SIZE_X_ * _NBL_GLSL_WORKGROUP_SIZE_Y_ * _NBL_GLSL_WORKGROUP_SIZE_Z_)
+
+layout (local_size_x = _NBL_GLSL_WORKGROUP_SIZE_X_, local_size_y = _NBL_GLSL_WORKGROUP_SIZE_Y_, local_size_z = _NBL_GLSL_WORKGROUP_SIZE_Z_) in;
+
+#include <../normalization/parameters.glsl>
+#include <../normalization/normalization.glsl>
+#include <../normalization/descriptors.glsl>
+// #include <../blit/formats/rgba16.glsl>
+
+layout (push_constant) uniform Block
+{
+	nbl_glsl_blit_normalization_parameters_t params;
+} pc;
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+nbl_glsl_blit_normalization_parameters_t nbl_glsl_blit_normalization_getParameters()
+{
+	return pc.params;
+}
+#define _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+#endif
```

Comment:
unifying this for all 3 shaders would be nice

Reply:
Resolved.